docs/design/security/README.adoc

```diff
@@ -0,0 +1,534 @@
+secure communication over the internet. It is the successor to the
+Secure Sockets Layer (SSL) protocol.
+TLS and SSL are similar in many ways, but TLS has some
+improvements over SSL, and it is now the standard protocol used for
+secure communication over the internet. The TLS configuration in
+Hazelcast is still named `ssl` for backward compatibility reasons.
+
+[source,yaml]
+----
+hazelcast:
+  network:
+    ssl:
+      enabled: true
+      properties:
+        # keystore contains a members identity
+        # (a private key and related certificate)
+        keyStoreType: PKCS12
+        keyStore: /opt/hazelcast-keystore.p12
+        keyStorePassword: YourSecretGoesHere
+        # truststore contains set of trusted certificates with public keys
+        trustStoreType: PKCS12
+        trustStore: /opt/hazelcast-truststore.p12
+        trustStorePassword: ACorrectPasswordGoesHere
+----
+
+Example how to generate the keystore and truststore:
+
+[source,bash]
+----
+# Generate a keystore with a key pair (including a self-signed certificate)
+keytool -genkeypair -keystore hazelcast-keystore.p12 -storetype PKCS12 \
+  -storepass YourSecretGoesHere -keypass YourSecretGoesHere \
+  -validity 7300 -keyalg EC -keysize 521 -sigalg SHA512withECDSA \
+  -alias member -dname "cn=member"
+
+# Export the self-signed certificate
+keytool -export -alias member -keystore hazelcast-keystore.p12 \
+  -storepass YourSecretGoesHere -file hazelcast.crt
+
+# Import the self-signed certificate to a new keystore/truststore
+keytool -import -keystore hazelcast-truststore.p12 -storetype PKCS12 \
+  -noprompt -alias member -storepass ACorrectPasswordGoesHere \
+  -file hazelcast.crt
+----
+
+[[data-at-rest-encryption]]
+== Data-at-rest encryption
+
+Hazelcast Persistence feature supports data encryption. It safeguards
+sensitive data stored on disk by encrypting it using industry-standard
+encryption algorithms. This feature helps protect data at rest from
+unauthorized access and enhances overall data security within the
+Hazelcast cluster.
+
+[source,yaml]
+----
+hazelcast:
+  map:
+    test-map:
+      data-persistence:
+        enabled: true
+  persistence:
+    enabled: true
+    base-dir: /mnt/persistence
+    backup-dir: /mnt/persistence-backup
+    encryption-at-rest:
+      enabled: true
+      algorithm: AES/CBC/PKCS5Padding
+      salt: NewYorkThymes
+      secure-store:
+        keystore:
+          path: /path/to/keystore.file
+          type: PKCS12
+          password: password
+----
+
+[[authentication-basics]]
+== Authentication
+
+Hazelcast supports authentication on Member protocol, Client protocol,
+and a subset of REST endpoints.
+
+Users can choose from a set of authentication mechanisms:
+
+* default (compares credentials against preconfigured member identity)
+* Simple - users and roles configured directly in the member
+configuration
+* TLS - available when mutual authentication is enabled
+* LDAP
+* Kerberos
```

Comment:
Will there be no explanation about these two in this document?

Reply:
Probably not. If I'll find some time in the future, I would like to prepare an extra document focused on LDAP authentication. (maybe also covering Kerberos).

For now, I'll add links to the official documentation for these topics.